lxd/events.go

```diff
@@ -35,7 +35,31 @@ func (r *eventsServe) String() string {
 }
 
 func eventsSocket(d *Daemon, r *http.Request, w http.ResponseWriter) error {
+	allProjects := queryParam(r, "all-projects") == "true"
+	projectQueryParam := queryParam(r, "project")
+	if allProjects && projectQueryParam != "" {
+		response.BadRequest(fmt.Errorf("'project' cannot be set if 'all-projects' is set"))
```

Comment:
We tend to avoid returning errors that reference specific API field names as they may not be familiar to the user who sees the error (if they are using the `lxc` tool or some other third party tool that uses the LXD API), as such this error should be rephrased to cover the concept that is wrong rather than the specific fields, e.g. something like:

"Cannot specify project when requesting a events for all projects".